packages/path_provider/path_provider_android/lib/path_provider_android.dart

```diff
@@ -81,7 +81,13 @@ class PathProviderAndroid extends PathProviderPlatform {
   }
 
   @override
-  Future<String?> getDownloadsPath() {
-    throw UnsupportedError('getDownloadsPath is not supported on Android');
+  Future<String?> getDownloadsPath() async {
+    final List<String>? paths =
+        await getExternalStoragePaths(type: StorageDirectory.downloads);
+    if (paths != null && paths.isNotEmpty) {
+      return paths.first;
+    } else {
+      throw UnsupportedError('getDownloadsPath is not supported on Android');
```

Comment:
Maybe change the error to something that references the empty path since it is now supported on Android?